File: hw/aymacana/u3/hw27GUIsOperation/ContactGUI_Java/src/main/java/utils/MongoConnection.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package utils;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

/**
 *
 * @author Mateo Cevallos
 */
public class MongoConnection {
    
    private static MongoDatabase dataBase;
    private static MongoClient mongoClient;
    
    public static MongoDatabase getConnection() {
        if (dataBase == null) {
            try {
                String uri = "mongodb+srv://Mateo:Mateo2006@cluster0.2mp0ve2.mongodb.net/?appName=Cluster0";

                mongoClient = MongoClients.create(uri);
                dataBase = mongoClient.getDatabase("TestDB");

                dataBase.runCommand(new Document("ping", 1));

            } catch (Exception e) {
                e.printStackTrace();
                dataBase = null;
            }
        }
        return dataBase;
    }

    public static boolean isConnected() {
        try {
            if (dataBase == null) {
                getConnection();
            }

            dataBase.runCommand(new Document("ping", 1));
            return true;

        } catch (Exception e) {
            return false;
        }
    }

    public static void closeConnection() {
        if (mongoClient != null) {
            mongoClient.close();
            dataBase = null;
            mongoClient = null;
        }
    }
    
}

File: exam/cevallos/u3/47-100/view/MongoConnection.js
const { MongoClient } = require('mongodb');

class MongoConnection {
    static instance = null;
    static client = null;
    static database = null;

    static async connect() {
        if (!MongoConnection.client) {
            try {
                const uri = 'mongodb+srv://Mateo:<db_password>@cluster0.2mp0ve2.mongodb.net/?appName=Cluster0';
                MongoConnection.client = new MongoClient(uri);

                await MongoConnection.client.connect();
                MongoConnection.database = MongoConnection.client.db('strategyCevallos');

                console.log('connected');
            } catch (error) {
                console.error('Error conecting to MongoDB:', error.message);
                throw error;
            }
        }
        return MongoConnection.database;
    }

    static async getCollection(collectionName) {
        if (!MongoConnection.database) {
            await MongoConnection.connect();
        }
        return MongoConnection.database.collection(collectionName);
    }

    static async close() {
        if (MongoConnection.client) {
            await MongoConnection.client.close();
            MongoConnection.client = null;
            MongoConnection.database = null;
            console.log('closed');
        }
    }

    static getStatus() {
        return {
            connected: !!MongoConnection.client,
            database: MongoConnection.database ? MongoConnection.database.databaseName : null
        };
    }
}

module.exports = MongoConnection;